src/Utils/CustomFieldIndexService.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace CubeTools\CubeCustomFieldsBundle\Utils;
+
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\Form\Form;
+
+/**
+ * This service allows to get required information from custom_fields table for building up index pages
+ * of entities linked to custom fields.
+ */
+class CustomFieldIndexService
+{
+    private $configReader;
+    private $em;
+
+    public function __construct(ConfigReader $configReader, EntityManager $em)
+    {
+        $this->configReader = $configReader;
+        $this->em = $em;
+    }
+
+    /**
+     * Retrieves the values of the custom fields for a given entity according to a given form fields set (filterform)
+     * Values are returned together with a raw parameter (boolean) indicating whether the raw filter should be applied during print out
+     *
+     * @param Form $filterform    Filter form containing all form fields which shall be available for filtering.
+     * @param object $entity    An entity stored in the database
+     *
+     * @return array            Contains the values of all custom fields which are referenced in the entity for all types contained in filterform, together with 'raw' indicator
+     */
+    public function getIndexRows(Form $filterform, $entity)
+    {
+        $entityClass = get_class($entity);
+        $fieldConfig = $this->configReader->getConfigForEntity($entityClass);
+        $indexRows = array();
+        // iterate over filterform custom fields
+        foreach ($filterform as $filterfield) {
+            if ($filterfield->getConfig()->getOption('translation_domain') == 'custom_fields') {
+                $fieldGetter = $filterfield->getName();
+                // we can access the customField directly by its name (using the magic getter function in the custom field trait):
+                $value = $entity->$fieldGetter;
+                $raw = false;
+                if ($fieldConfig[$fieldGetter]['type'] == 'Ivory\CKEditorBundle\Form\Type\CKEditorType') {
+                    $raw = true;
+                }
+                if (is_object($value) && $value instanceof \DateTimeInterface) {
+                    $value = $value->format('d.m.Y');
```

Comment:
Should this not be handled in the template? Like this it is impossible to show time.
To detect DateTime see https://github.com/EmchBerger/projectassistant/commit/ffcc99a990ef3b26e4e13807290955a873c35b71

Or add some different option to configure time format column wise.

Reply:
agree that this is not the best solution currently. since it's copy paste (or cut paste) from reservations, I suggest we leave it this way for now. if we need date and time in the future, we can then add an improvement (maybe optional parameter or something)